### ISSUE TITLE
Openhab2 install via apt-get

### ISSUE THREAD

**Issue by stefaanbolle** [openhab/openhab-distro#66]
First of all a big thank you for the excellent work on Openhab!
I have a running 1.8.0 version with a lot of bindings and I want to migrate to 2.0 beta1.

To ease installation, would it be possible to make a debian repository available (like 1.8.0) with the beta release and bindings to install the latest release via apt-get on my raspberry pi2?

**Comment by kaikreuzer** [comment]
Yes, offering debian packages is definitely planned, but not yet available.
This is probably the right moment to ask @theoweiss to start looking into this :-)

**Comment by kaikreuzer** [comment]
FTR: First skeleton in https://github.com/openhab/openhab-distro/pull/73

**Comment by kaikreuzer** [comment]
To clarify:
There should be imho only a single debian package created for the distro (or at most 2 packages: one for the online distro and one for the offline distro).
Add-ons should NOT be managed as apt packages anymore, since this is now all handled through Karaf (which also has the information about dependencies between bundles etc.).

**Comment by stefaanbolle** [comment]
@kaikreuzer 
Does this mean that all we have to do is install the distro and karaf will take care of getting the addon jars (online) or already has the addon jars (offline) in its distro package?

**Comment by kaikreuzer** [comment]
Yes, the selection of addons is now merely a configuration entry, see https://github.com/openhab/openhab-distro/blob/master/docs/sources/getting-started.md#choosing-a-base-package-and-add-ons-to-be-installed

**Comment by stefaanbolle** [comment]
nikobus binding is not in the list.
Can I still copy the binding jar to ./addons?

**Comment by dvanherbergen** [comment]
You can try. It has not been tested with 2.0 yet.

**Comment by kaikreuzer** [comment]
See https://github.com/openhab/openhab-distro/blob/master/docs/sources/features/compatibilitylayer.md#how-to-use-openhab-1x-add-ons-that-are-not-part-of-the-distribution

**Comment by stefaanbolle** [comment]
I have touched a lot of wood :-) but a lot of my addons are currently not running smooth.
Openhab2 does not give any error on the nikobus binding, but my light channel doesn't respond to the opnehab2 switch action to turn it on or off.
a lot of trial and error and troubleshooting coming ahead :-)
Btw this becomes way to far off topic.
Thanks Kai for your assistance!

**Comment by kaikreuzer** [comment]
> Openhab2 does not give any error on the nikobus binding

Always enable debug logging (see "log:set" as shown [here](https://github.com/openhab/openhab-distro/blob/master/docs/sources/getting-started.md#using-the-shell)) when you want detailed logs.

> Btw this becomes way to far off topic.

You are right, let's stop here and continue in the forum if necessary!

**Comment by theoweiss** [comment]
@kaikreuzer I've almost overlooked this. Ok, I will start to work on the packaging things.

**Comment by drock1985** [comment]
Hi @kaikreuzer 

Did you have a chance to get the OH2 beta in repository? Would love to test this out on a spare RPI that I have laying around.

**Comment by kaikreuzer** [comment]
@drock1985 Why do you need an apt package for testing?

**Comment by drock1985** [comment]
Hi @kaikreuzer 

Don't need it, but in the end i'll be using apt-get as to install OH2 when it comes out of beta. Might as well test what I am sure will be the preferred install method right from the start, to see if there will be any problems.

**Comment by kaikreuzer** [comment]
øk, then just wait for news from @theoweiss on this issue, who is working on it.

**Comment by skatun** [comment]
Any release date set for the apt-get method for OH2?

**Comment by mikef2501** [comment]
I'm interested, too!  Is there any sort of update or timeline?  
Thanks a bunch!

**Comment by kaikreuzer** [comment]
Just follow https://github.com/openhab/openhab-distro/pull/138 and help testing it!

**Comment by kaikreuzer** [comment]
FTR: https://github.com/openhab/openhab-distro/pull/213 has just been merged, so that the build now produces Debian packages. It would be great if you could go testing these and I will check with @theoweiss to soon put a version on Bintray as well.

**Comment by theoweiss** [comment]
You can test the current __snapshot__ of the offline distribution by adding this line to your apt-sources:
`deb https://openhab.ci.cloudbees.com/job/openHAB-Distribution/ws/distributions/openhab-offline/target/apt-repo/ /`
or the online distribution by adding that line:
`deb https://openhab.ci.cloudbees.com/job/openHAB-Distribution/ws/distributions/openhab-online/target/apt-repo/ /
`

**Comment by mckennajp** [comment]
apt install working great so far.  On Ubuntu, I had to install apt-transport-https first (apt-get install apt-transport-https).  Thanks a lot!

**Comment by mikef2501** [comment]
I wiped the SD card on my Raspberry Pi 2 and did a clean install of Raspbian.  After adding the apt-sources line for openhab-online, it required me to install apt-transport-https like mckennajp.  Even though I did this and added the apt-key for openhab, it still gave me a warning that it couldn't authenticate and asked me if I wanted to continue without authentication.

I haven't had much time between work and kids, but so far it appears to have installed correctly.  I installed the Phillips Hue module through the Paper GUI, then the base and lights, and can control the lights just fine.  

I'll post more when I am able to spend some more time.  Thanks!

**Comment by theoweiss** [comment]
Thanks for testing. Good news so far.
The snapshot repo on cloudbees is not signed the upcoming bintray repo for the releases will be signed.

**Comment by abedwardsw** [comment]
I tested the openhab2-offline deb pkg as well, seems to work pretty good!  Thanks for making this, will make updates much easier.  Any thoughts on including habmin snapshot jar file as well?  You could put it in a unused directory, then people who want it could put a symlink in addons, this way it's not on by default but it would get updated with the deb update.  Not sure if there is plans to add it as a karaf bundle or not?

Probably unrelated, but when I used the offline version, the first startup I got the following error, which went away after a restart.  Wanted to point it out just in case.
```
2016-05-29 14:22:36.235 [ERROR] [.glassfish.hk2.osgi-resource-locator] - FrameworkEvent ERROR - org.glassfish.hk2.osgi-resource-locator
org.osgi.framework.BundleException: Exception in org.apache.karaf.features.internal.service.FeaturesServiceImpl$3.end()
        at org.eclipse.osgi.internal.framework.OSGiFrameworkHooks$CoreResolverHookFactory.handleHookException(OSGiFrameworkHooks.java:164)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.internal.framework.OSGiFrameworkHooks$CoreResolverHookFactory$CoreResolverHook.end(OSGiFrameworkHooks.java:340)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleResolver$ResolveProcess.resolve(ModuleResolver.java:901)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleResolver.resolveDelta(ModuleResolver.java:111)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer.resolveAndApply(ModuleContainer.java:479)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:437)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer.resolve(ModuleContainer.java:427)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.Module.start(Module.java:416)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1498)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)[org.eclipse.osgi-3.10.2.v20150203-1939.jar:]
2016-05-29 14:22:55.181 [ERROR] [core.karaf.internal.FeatureInstaller] - Failed installing 'openhab-package-standard': Error restarting bundles
2016-05-29 14:22:55.206 [ERROR] [core.karaf.internal.FeatureInstaller] - Failed installing 'openhab-package-standard': Error restarting bundles
2016-05-29 14:28:06.977 [ERROR] [org.eclipse.smarthome.io.rest       ] - FrameworkEvent ERROR - org.eclipse.smarthome.io.rest
java.lang.NoClassDefFoundError: org/eclipse/smarthome/io/transport/mdns/ServiceDescription
        at java.lang.Class.getDeclaredMethods0(Native Method)[:1.8.0_65]
        at java.lang.Class.privateGetDeclaredMethods(Class.java:2701)[:1.8.0_65]
        at java.lang.Class.getDeclaredMethods(Class.java:1975)[:1.8.0_65]
```

**Comment by kaikreuzer** [comment]
@abedwardsw I have seen such errors sometimes on a the first startup of a normal install as well - so this is not related to the apt installation.

**Comment by maggu2810** [comment]
This is caused by the usage of optional resolution and so really not related to the Debian packages.

**Comment by msth** [comment]
If the last build fails, than the download fails. Can we use someting like

> deb https://openhab.ci.cloudbees.com/job/openHAB-Distribution/lastSuccessfulBuild/artifact/distributions/openhab-offline/target/apt-repo/ /

**Comment by modestos** [comment]
I personally tested the openhab2-online deb pkg on a beaglebone black with minimal debian installation on internal storage. It boots quite fast and starting service was snappy. 

It works flawlessly, thx.

**Comment by fharni** [comment]
Hi, I have a problem with the apt repository. As descriped I added the repository and does a apt-get update. This complains about a missing Packages entry in the release file.

`W: Herunterladen von https://openhab.ci.cloudbees.com/job/openHAB-Distribution/ws/distributions/openhab-offline/target/apt-repo/Release fehlgeschlagen: Erwarteter Eintrag »Packages« konnte in Release-Datei nicht gefunden werden (falscher Eintrag in sources.list oder missgebildete Datei).`
`W: Herunterladen von https://openhab.ci.cloudbees.com/job/openHAB-Distribution/ws/distributions/openhab-online/target/apt-repo/Release fehlgeschlagen: Erwarteter Eintrag »Packages« konnte in Release-Datei nicht gefunden werden (falscher Eintrag in sources.list oder missgebildete Datei).`
`E: Manche Indexdateien konnten nicht heruntergeladen werden. Sie wurden ignoriert oder durch alte ersetzt.`

**Comment by jellebens** [comment]
I get:
W: GPG error: https://openhab.ci.cloudbees.com  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY F834FE4EEAC1183A
W: GPG error: https://openhab.ci.cloudbees.com  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY F834FE4EEAC1183A

Where can I find the public key?

**Comment by ThomDietrich** [comment]
@fharni your sources file seems to be wrong. Please follow the instructions given here: http://docs.openhab.org/installation/linux.html
